src/utils.c

```diff
@@ -142,6 +142,136 @@ ts_time_value_to_internal(Datum time_val, Oid type_oid, bool failure_ok)
 	}
 }
 
+TSDLLEXPORT int64
+ts_interval_value_to_internal(Datum time_val, Oid type_oid, bool failure_ok)
+{
+	switch (type_oid)
+	{
+		case INT8OID:
+		case INT4OID:
+		case INT2OID:
+			return ts_integer_to_internal(time_val, type_oid, failure_ok);
```

Comment:
this is only taken for INT8/4/2OIDs, even though ts_integer_to_internal can also handle types that are binary compatible with int8. I think it'd be good to add support for binary compatible types here as well. That'd mean that the case statement needs to be rewritten slightly to call ts_integer_to_internal in the default case